activerecord/lib/active_record/connection_adapters/postgresql/oid/specialized_string.rb

```diff
@@ -5,12 +5,18 @@ module ConnectionAdapters
     module PostgreSQL
       module OID # :nodoc:
         class SpecializedString < Type::String # :nodoc:
+          CASE_INSENSITIVE_TYPES = [:macaddr]
```

Comment:
This might be better implemented as a separate `PostgreSQL::OID::Macaddr` type.  For reference, there are already [`PostgreSQL::OID::Cidr`](https://github.com/rails/rails/blob/72d293601346f708082d9ecfcd9b99b987f64783/activerecord/lib/active_record/connection_adapters/postgresql/oid/cidr.rb) and [`PostgreSQL::OID::Inet`](https://github.com/rails/rails/blob/72d293601346f708082d9ecfcd9b99b987f64783/activerecord/lib/active_record/connection_adapters/postgresql/oid/inet.rb) types.

Reply:
Sounds good